src/App.tsx

```diff
@@ -1,223 +1,259 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useState,
+  useContext,
+  useEffect,
+  useRef,
+  useMemo,
+  useCallback,
+} from 'react';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorMessages } from './components/ErrorMessages';
+import { TodoContent } from './components/TodoContent/TodoContent';
+
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+import { ErrorType } from './types/ErrorType';
+
+import {
+  getTodos,
+  createTodo,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [proccessedTodoId, setProccessedTodoId] = useState<number[]>([]);
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const isEveryTodosComplited = useMemo(() => (
+    todos.every(todo => todo.completed)
+  ), [todos]);
+
+  const countOfTodos = useMemo(() => todos.length, [todos]);
```

Comment:
it's redundant to cover primitive value with useMemo
in case with objects or functions we need it, because for each rerendering we receive the new link for some object
but in case with primitive values, the value doesn't change (there is no any link changes), it's still the same primitive value, so you don't need to use useMemo for it